### PR TITLE
CGM from systems

### PR DESCRIPTION
**Pull request by profxj**
Generates CGM objects from absorption systems and galaxies.

Tests but no rst docs.

### CODE REVIEW

File: pyigm/cgm/utils.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    # Cosmology
+    if cosmo is None:
+        cosmo = cosmology.Planck15
+
+    # R
+    rho, angles = calc_rho(galaxy, igmsystems, cosmo)
+
+    # dv
+    igm_z = np.array([igmsystem.zabs for igmsystem in igmsystems])
+    dv = ltu.v_from_z(galaxy.z, igm_z)
```
Comment (jnburchett) [inline pyigm/cgm/utils.py]: I get a DeprecationWarning suggesting we "use dv_from_z() instead".

File: pyigm/igm/igmsightline.py [hunk 2]
```diff
@@ -11,12 +11,32 @@
...
 class IGMSightline(AbsSightline):
     """Class for IGM Absorption Sightline
     """
     @classmethod
+    def from_json(cls, jsonfile, coord=None, **kwargs):
+        """ Instantiate from a JSON file
+
+        Parameters
+        ----------
+        jsonfile
```
Comment (jnburchett) [inline pyigm/igm/igmsightline.py]: Do the jsonfile contents adhere to some particular format?  Might be helpful to list required quantities defined within the file (RA, DEC, zem, name).

File: pyigm/cgm/tests/test_init_cgm.py [hunk 3]
```diff
@@ -55,3 +58,14 @@ def test_to_dict():
...
+
+def test_cgm_from_igmsystems():
+    # Load sightlines
+    sl_file = pyigm.__path__[0]+'/data/sightlines/Blind_CIV/J115120.46+543733.08.json'
+    igmsl = IGMSightline.from_json(sl_file)
+    igmsys = igmsl.make_igmsystems(vsys=400*u.km/u.s)
+    # Galaxy
+    galaxy = Galaxy((178.84787, 54.65734), z=0.00283)
+    # Go
+    cgm_list = cgm_from_galaxy_igmsystems(galaxy, igmsys, chk_lowz=False)
```
Comment (jnburchett) [inline pyigm/cgm/tests/test_init_cgm.py:70]: I really need to knock out this chk_lowz feature...

Comment (jnburchett) [reply pyigm/cgm/tests/test_init_cgm.py:70]: It turns out the DeprecationWarning doesn't allow the instantiation.

**Review by jnburchett — CHANGES_REQUESTED** [review]

File: pyigm/cgm/utils.py (hunk 1 above)
Comment (profxj) [reply pyigm/cgm/utils.py]: Fixed.

File: pyigm/igm/igmsightline.py (hunk 2 above)
Comment (profxj) [reply pyigm/igm/igmsightline.py]: added to docs

File: pyigm/cgm/tests/test_init_cgm.py (hunk 3 above)
Comment (profxj) [reply pyigm/cgm/tests/test_init_cgm.py:70]: Test modified.  Passing on my laptop.

**Review by jnburchett — APPROVED** [review]

**Comment by jnburchett** [comment]
For some reason, a Python 3.4 check is failing on test_make_igmsystems().  I am still running Python 2.7, so I of course do not get the AssertionError found above.  Please merge if you can live with this.

**Comment by profxj** [comment]
Passing tests now.  Needed the linetools update.
Merging.

**Pull request merged**